frontend/src/pages/components/Navbar/DesktopNavigation.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState, useEffect } from "react";
+import Link from "next/link";
+import Image from "next/image";
+import {
+  AppsSection,
+  NavLink,
+  ResourceSection,
+  Social,
+} from "@/queries/navbar";
+import DownArrowIcon from "@/assets/svgs/icons/down-arrow.svg";
+import AppsDropdownContent from "./AppsDropdownContent";
+import ResourcesDropdownContent from "./ResourcesDropdownContent";
+
+interface DesktopNavigationProps {
+  pathname: string;
+  navLinks: NavLink[];
+  appsSection: AppsSection;
+  resourceSections: ResourceSection[];
+  socials: Social[];
+}
+
+const DesktopNavigation: React.FC<DesktopNavigationProps> = ({
+  pathname,
+  navLinks,
+  appsSection,
+  resourceSections,
+  socials,
+}) => {
+  const [openDropdownIndex, setOpenDropdownIndex] = useState<number | null>(
+    null
+  );
+
+  useEffect(() => {
+    if (openDropdownIndex !== null) {
+      document.body.style.overflow = "hidden";
+    } else {
+      document.body.style.overflow = "";
+    }
+  }, [openDropdownIndex]);
```

Comment:
I would add `react-use` package since it has a builtin hook for this: https://github.com/streamich/react-use/blob/master/docs/useLockBodyScroll.md